src/openms/include/OpenMS/CHEMISTRY/ModificationsDB.h

```diff
@@ -146,9 +147,12 @@ namespace OpenMS
 
     /**
        @brief Add a new modification to ModificationsDB.
-       Will skip adding if modification already exists (based on its fullID)
+       If the modification already exists (based on its fullID) it is not added.
+       The function returns a pointer to the modification in the ModificationDB (which can be differ from input if mod was already present).
+
+       @param new_mod Owning pointer, which transfers ownership to ModificationsDB (mod might get deleted if already present!)
     */
-    void addModification(ResidueModification * new_mod);
+    const ResidueModification* addModification(std::unique_ptr<ResidueModification> new_mod);
```

Comment:
does std::unique_ptr work with python?

Reply:
there was no binding for `addModification` before, so I assume it did not even work with pointers.